lib/conjur/puppet_module/http.rb

```diff
@@ -1,50 +1,50 @@
 # frozen_string_literal: true
 
-require 'conjur/puppet_module/ssl'
+require_relative 'ssl'
 
 module Conjur
   module PuppetModule
     # This module is in charge of interacting with the Conjur endpoints
     module HTTP
       class << self
         def get(host_url, path, ssl_certificate, token)
-          uri, use_ssl = parse_url(host_url, path)
-          certs = Conjur::PuppetModule::SSL.load(ssl_certificate)
-
           headers = {}
           if token
             encoded_token = Base64.urlsafe_encode64(token)
             headers['Authorization'] = "Token token=\"#{encoded_token}\""
           end
 
-          Net::HTTP.start uri.host, uri.port, use_ssl: use_ssl, cert_store: certs do |http|
-            response = http.get(uri.request_uri, headers)
+          start_http(host_url, path, ssl_certificate) do |http, uri|
+            http.get(uri.request_uri, headers)
+          end
+        end
 
-            raise Net::HTTPError.new response.message, response unless response.code.match?(%r{^2})
+        def post(host_url, path, ssl_certificate, data, token = nil)
+          headers = {}
+          if token
+            headers['Authorization'] = "Token token=\"#{token}\""
+          end
 
-            response.body
+          start_http(host_url, path, ssl_certificate) do |http, uri|
+            http.post(uri.request_uri, data, headers)
           end
         end
 
-        def post(host_url, path, ssl_certificate, data)
-          uri, use_ssl = parse_url(host_url, path)
-
-          raise(ArgumentError, "POST data to #{uri} must not be empty!") \
-            if data.nil? || data.empty?
+        private
 
+        def start_http(host_url, path, ssl_certificate)
```

Comment:
`s/start_http/start_http_request` might help to be more explicit. At first glance I wasn't sure if `start_http` was creating a mock server or making request.